python/pyspark/sql/dataframe.py

```diff
@@ -70,12 +70,12 @@ class DataFrame(object):
     .. versionadded:: 1.3
     """
 
-    def __init__(self, jdf, sql_ctx):
+    def __init__(self, jdf, sql_ctx, schema=None):
         self._jdf = jdf
         self.sql_ctx = sql_ctx
         self._sc = sql_ctx and sql_ctx._sc
         self.is_cached = False
-        self._schema = None  # initialized lazily
+        self._schema = schema  # initialized lazily if None
```

Comment:
@BryanCutler, what do you think about taking this out back? Maybe, I am too much worried but I think we maybe should avoid it to be assigned actually except for few special cases ...

Reply:
Sure, I can undo it.  I don't really like manually assigning the schema after the constructor but it's just done in these 2 special cases..